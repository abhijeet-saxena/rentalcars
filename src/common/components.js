import styled from 'styled-components';

export const colors = {
    primaryBlue: '#1E79CA',
    secondaryGreen: '#3D8425',
    secondaryYellow: '#F7B732',
    textDark: '#333',
    textLight: '#FFF',
    textBlue: '#1273C4',
    backgroundGray: '#f5f5f5',
};

export const Flex = styled.div`
    display: flex;
    flex-direction: ${(props) => props.flexDirection || 'row'};
    padding: ${(props) => props.padding || '0'};
    margin: ${(props) => props.margin || '0'};
    width: ${(props) => props.width || '100%'};
    justify-content: ${(props) => props.justifyContent || 'normal'};
    align-items: ${(props) => props.alignItems || 'normal'};
    background: ${(props) => props.background || colors.textLight};
    text-align: ${(props) => props.textAlign || 'left'};
    font-size: ${(props) => props.fontSize || '1rem'};
    color: ${(props) => props.color || colors.textDark};
`;

export const Box = styled.div`
    padding: ${(props) => props.padding || '0'};
    margin: ${(props) => props.margin || '0'};
    width: ${(props) => props.width || '100%'};
    font-size: inherit;
`;

export const Link = styled.a`
    text-decoration: ${(props) => (props.underline ? 'underline' : 'none')};
    color: ${(props) => props.color || colors.textLight};
    padding: ${(props) => props.padding || '0'};
    font-size: inherit;
    &:hover {
        text-decoration: underline;
    }
`;
