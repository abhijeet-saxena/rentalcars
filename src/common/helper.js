export const togglePanel = ({ target }) => {
    target.classList.toggle('expanded');
    let content = target.nextElementSibling;

    if (content.style.height) {
        content.style.height = null;
        content.style.margin = '0px';
    } else {
        content.style.height = `${content.scrollHeight + 16}px`;
        content.style.margin = '0 1rem 1rem';
    }
};

export const debounce = (f, delay = 250) => {
    let timer;

    return function (...args) {
        let that = this;

        clearTimeout(timer);
        timer = setTimeout(() => {
            f.apply(that, [...args]);
        }, delay);
    };
};
