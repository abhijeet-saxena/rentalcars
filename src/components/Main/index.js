import React from 'react';
// import { Flex, Box, Link } from '../../common/components';
import { MainContainer } from './styles';

const Main = () => {
    return (
        <MainContainer>
            <h1>Abhijeet</h1>
        </MainContainer>
    );
};

export default Main;
