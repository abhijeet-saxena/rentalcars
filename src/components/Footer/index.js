import React from 'react';
import { Box, Link } from '../../common/components';
import { FooterContainer, LinksContainer, ImagesContainer } from './styles';

import booking from '../../assets/images/booking.png';
import priceline from '../../assets/images/priceline.png';
import kayak from '../../assets/images/kayak.png';
import agoda from '../../assets/images/agoda.png';
import rentalcars from '../../assets/images/rentalcars.png';
import opentable from '../../assets/images/opentable.png';

const Footer = () => {
    return (
        <FooterContainer>
            <LinksContainer>
                <Link href="#" padding="0.5rem 0">
                    Privacy Notice
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Cookies
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Terms & Conditions
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Help
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Modern Slavery Statement
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Supply Partner Enquiry and Marketplace
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Affiliate Programme
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Careers
                </Link>
            </LinksContainer>
            <Box width="80%" margin="1rem 0">
                Rentalcars.com is a trading name of Booking.com Transport
                Limited which is a limited company registered in England and
                Wales (Number: 05179829) whose registered address is at 100 New
                Bridge Street, London, EC4V 6JA. VAT number: GB 855349007.
                <br />
                <br />
                Rentalcars.com is part of Booking Holdings Inc., the world
                leader in online travel &amp; related services.
                <br />
                <br />
                Booking.com Transport Limited (FCA Firm Reference No 777648) is
                an Appointed Representative of Cover Genius Limited, which is
                authorised and registered by the Financial Conduct Authority,
                Firm Reference No 750711. Traveljigsaw Insurance Limited is
                incorporated in Malta and is authorised by the Malta Financial
                Services Authority to carry on the business of insurance in
                terms of the Insurance Business Act 1998. Company Registration
                Number: C82173 Registered Office, Development House, St Anne
                Street, Floriana, FRN 9010, Malta.
                <br />
                <br />
                <ImagesContainer>
                    <img src={booking} alt="booking" width="91px"></img>
                    <img src={priceline} alt="priceline" width="91px"></img>
                    <img src={kayak} alt="kayak" width="79px"></img>
                    <img src={agoda} alt="agoda" width="70px"></img>
                    <img src={rentalcars} alt="rentalcars" width="109px"></img>
                    <img src={opentable} alt="opentable" width="95px"></img>
                </ImagesContainer>
                <br />
                Copyright Â© 2020 Booking.com Transport Limited trading as
                Rentalcars.com. All rights reserved
            </Box>
        </FooterContainer>
    );
};

export default Footer;
