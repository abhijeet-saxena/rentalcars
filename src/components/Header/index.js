import React from 'react';
import { Flex } from '../../common/components';
import { colors } from '../../common/helper';
import { HeaderContainer } from './styles';
import Navigation from './navigation';
import Search from './search';

const Header = () => {
    return (
        <Flex background={colors.primaryBlue} justifyContent="center">
            <HeaderContainer data-testid="heading">
                <Navigation></Navigation>
                <h1>Car Hire â€“ Search, Compare &amp; Save</h1>
                <Search></Search>
            </HeaderContainer>
        </Flex>
    );
};

export default Header;
