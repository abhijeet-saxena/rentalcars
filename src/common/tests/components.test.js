import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Flex, Box, Link, Badge, SuggestionCard, Drawer } from '../components';
import { colors } from '../helper';

afterEach(cleanup);

it('Rendered Snapshot Matches', () => {
    const flex = render(
        <Flex>
            <h1>lorem ipsum dolet</h1>
        </Flex>,
    );
    const box = render(
        <Box>
            <p>lorem ipsum dolet</p>
        </Box>,
    );
    const link = render(<Link href="#">lorem ipsum dolet</Link>);
    const badge = render(<Badge bg={colors.badgeBlue}></Badge>);
    const drawer = render(
        <Drawer closeMenu={() => {}} heading="Menu">
            <div>Children Element</div>
        </Drawer>,
    );
    const card = render(
        <SuggestionCard
            details={{
                city: 'Sydney',
                country: 'Australia',
                name: 'Sydney Airport',
                region: 'New South Wales',
                placeType: 'A',
                iata: 'SYD',
            }}
            setValue={() => {}}
        ></SuggestionCard>,
    );

    expect(flex).toMatchSnapshot();
    expect(box).toMatchSnapshot();
    expect(link).toMatchSnapshot();
    expect(badge).toMatchSnapshot();
    expect(drawer).toMatchSnapshot();
    expect(card).toMatchSnapshot();
});
