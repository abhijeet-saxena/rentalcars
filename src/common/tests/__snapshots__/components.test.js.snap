// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered Snapshot Matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM hoHvTM"
    >
      <h1>
        lorem ipsum dolet
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendered Snapshot Matches 2`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtsrHT dWcgVK"
    >
      <p>
        lorem ipsum dolet
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendered Snapshot Matches 3`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <a
      class="sc-dlnjwi ejjUYJ"
      href="#"
    >
      lorem ipsum dolet
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendered Snapshot Matches 4`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendered Snapshot Matches 5`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
      data-testid="drawer"
    >
      <div
        class="sc-bdnxRM jPJVms header"
      >
        <div>
          Menu
        </div>
        <button
          aria-label="Close Menu"
          type="button"
        >
          <img
            alt=""
            src="dismiss.svg"
          />
        </button>
      </div>
      <hr />
      <div>
        Children Element
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Rendered Snapshot Matches 6`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hoHvTM"
      >
        <h1>
          lorem ipsum dolet
        </h1>
      </div>
    </div>
    <div>
      <div
        class="sc-gtsrHT dWcgVK"
      >
        <p>
          lorem ipsum dolet
        </p>
      </div>
    </div>
    <div>
      <a
        class="sc-dlnjwi ejjUYJ"
        href="#"
      >
        lorem ipsum dolet
      </a>
    </div>
    <div>
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK bMwgbw"
      />
    </div>
    <div>
      <div
        class="sc-bdnxRM sc-eCApnc hoHvTM bIYIYn"
        data-testid="drawer"
      >
        <div
          class="sc-bdnxRM jPJVms header"
        >
          <div>
            Menu
          </div>
          <button
            aria-label="Close Menu"
            type="button"
          >
            <img
              alt=""
              src="dismiss.svg"
            />
          </button>
        </div>
        <hr />
        <div>
          Children Element
        </div>
      </div>
    </div>
    <div>
      <div
        class="sc-bdnxRM jjvUKX"
        data-testid="drawer"
      >
        <div
          class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
        >
          Airport
        </div>
        <div>
          <p>
            Sydney Airport
             
            (SYD)
          </p>
          <small>
            Sydney, 
             
            New South Wales, 
             
            Australia
          </small>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM jjvUKX"
      data-testid="drawer"
    >
      <div
        class="sc-gtsrHT sc-hKFxyN dWcgVK dHQebd"
      >
        Airport
      </div>
      <div>
        <p>
          Sydney Airport
           
          (SYD)
        </p>
        <small>
          Sydney, 
           
          New South Wales, 
           
          Australia
        </small>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
